client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerLeakTest.java

```diff
@@ -38,7 +41,10 @@ public void call() {
             eventListener.expectOnlyCreatedEvent(1, cache(cacheName));
          }
       });
-      detectThreadLeaks(".*Client-Listener-ClientListenerLeakTest-.*");
+      WaitFor
+              .condition(() -> listThreadsMatchingRegexp(".*Client-Listener-ClientListenerLeakTest-.*").size() == 0)
+              .atMost(10, TimeUnit.MILLISECONDS)
+              .await();
```

Comment:
According to https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html#shutdown  "This method does not wait for previously submitted tasks to complete execution. Use awaitTermination to do that."